usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -10387,6 +10404,18 @@ CK_RV ep11tok_is_mechanism_supported(STDLL_TokData_t *tokdata,
             goto out;
         }
         break;
+
+    case CKM_AES_XTS:
+    case CKM_AES_XTS_KEY_GEN:
+        if (ep11tok_pkey_option_disabled(tokdata) || ep11_data->msa_level < 4 ||
+            ep11tok_is_mechanism_supported(tokdata, CKM_IBM_CPACF_WRAP) != CKR_OK ||
+            ep11tok_is_mechanism_supported(tokdata, CKM_AES_KEY_GEN) != CKR_OK) {
```

Comment:
Maybe nothing for this pr, but we could change the logic of ep11tok_is_mechanism_supported() to return either 'true' or 'false', as the function name suggests. Currently the function either returns 0 if the mech is supported or CKR_MECHANISM_INVALID if it's not. So already only two return values. It would make the usage of the function more self-explaining.

Reply:
Right, nothing for this PR. The intention of the return code of ep11tok_is_mechanism_supported() was that it could further be returned directly by the caller, as it is for example done in ep11tok_get_mechanism_info().

And actually, ep11tok_is_mechanism_supported() does return other return codes than CKR_MECHANISM_INVALID, namely CKR_FUNCTION_FAILED. Returning a bool would not allow to distinguish between a real error case and the 'not-supported' case.